x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/core/security/transport/netty4/SecurityNetty4Transport.java

```diff
@@ -86,10 +88,19 @@ public SecurityNetty4Transport(
         this.exceptionHandler = new SecurityTransportExceptionHandler(logger, lifecycle, (c, e) -> super.onException(c, e));
         this.sslService = sslService;
         this.transportSslEnabled = XPackSettings.TRANSPORT_SSL_ENABLED.get(settings);
-        this.remoteClusterSslEnabled = REMOTE_CLUSTER_PORT_ENABLED.get(settings) && REMOTE_CLUSTER_SSL_ENABLED.get(settings);
+        final boolean remoteClusterClientSslEnabled = REMOTE_CLUSTER_SSL_ENABLED.get(settings);
+        this.remoteClusterServerSslEnabled = REMOTE_CLUSTER_PORT_ENABLED.get(settings) && remoteClusterClientSslEnabled;
 
-        this.profileConfiguration = Collections.unmodifiableMap(ProfileConfigurations.get(settings, sslService, true));
-        this.sslConfiguration = this.profileConfiguration.get(TransportSettings.DEFAULT_PROFILE);
+        this.profileConfigurations = Collections.unmodifiableMap(ProfileConfigurations.get(settings, sslService, true));
+        this.defaultSslConfiguration = this.profileConfigurations.get(TransportSettings.DEFAULT_PROFILE);
+        assert this.transportSslEnabled == false || this.defaultSslConfiguration != null;
+
+        if (remoteClusterClientSslEnabled) {
+            this.remoteClusterClientSslConfiguration = sslService.getSSLConfiguration(XPackSettings.REMOTE_CLUSTER_SSL_PREFIX);
```

Comment:
This part needs some explanation. We have an issue for handling client side (QC) SSL configuration due to the facts:
1. `_remote_cluster` is not truly a reserved profile name. If remote cluster port is enabled, it is reserved because we throw error if direct usage is detected. But if the port is disabled (by default), this is just a regular profile name.
2. Remote cluster SSL is enabled by default. 

The above works well for the server side (FC): If the port is disabled, no SSL configuration is needed. User can directly configure `_remote_cluster` profile to have SSL. But that is completely seperate from the new Remote Cluster feature. If the port is enabled, we force users to choose between either not using the new RCS or removing the direct usage of `_remote_cluster`. 

But on the client side (QC), we don't want to tie SSL with whether the port is enabled. If the cluster is simple a QC, it is unreasonable to ask it enable the port before it can enable SSL for connecting a remote cluster. Therefore, we want build the SSL configuration at all times for remote cluster. This is the problem. Because the port may be disabled, which means direct usage of `_remote_cluster` profile is *not* prevented. This in turns means we cannot use the `_remote_cluster` profile name as the key for remote cluster SSL configuration because it may already be occupied by user's directly usage of the same profile. Concretely, it means the following configuration must keep working and at the meantime support the cluster being a new FC:

```yaml
transport.profiles._remote_cluster.xpack.security.ssl.key: xxx
transport.profiles._remote_cluster.xpack.security.ssl.certificate: xxx
```

The `profileConfigurations` map will have an entry with `_remote_cluster` and value is the direct configuration of the profile. We can no longer insert our remote cluster entry into the map because of key conflicts. So instead, I keep the client side remote cluster SSL configuration on its own to avoid touch the map.

Alternatively, we can make SSL default to disabled for remote cluster. So if user explicitly opt-in for enabling remote cluster SSL, we can throw error when we detect direct usage of the profile. But I feel it is too much a tradeoff, i.e. we reduce the level of security because of an implementation difficulty.